zapier/cradlApi.js

```diff
@@ -42,84 +42,70 @@ async function putToFileServer(z, url, content) {
-  const postDocumentsResponse = await makePostRequest(z, '/documents', {})
-  // bundle.inputData.file will be a URL from which we download the file
-  fileResponse = await downloadFile(inputFileUrl, z)
-  const fileServerResponse = await putToFileServer(z, postDocumentsResponse.json.fileUrl, fileResponse.buffer())
-  return postDocumentsResponse.json.documentId
-}
-
-async function createPrediction(z, documentId, modelId){
-  return makePostRequest(z, '/predictions', {
-    documentId: documentId,
-    modelId: modelId,
-    postprocessConfig: {
-      outputFormat: 'v2',
-      parameters: {
-        n: 3,
-        collapse: true,
-      },
-      strategy: 'BEST_N_PAGES',
-    }
+async function getFromFileServer(z, url, raw=false) {
+  if (url == null) {
+    return {}
+  }
+  
+  return z.request({
+    url: url,
+    method: 'GET',
+    raw: raw,
   });
 }
 
-async function listModels(z) {
-  return makeGetRequest(z, '/models?owner=me&owner=las:organization:cradl')
+async function getAction(z, actionId) {
+  return makeGetRequest(z, '/actions/' + actionId)
 }
 
-async function executeWorkflow(z, documentId, title, workflowId, metadata){
-  input = {
-    documentId: documentId,
-    title: title,
-  }
-  if (metadata) {
-    input.metadata = metadata
-  }
-  return makePostRequest(z, '/workflows/' + workflowId + '/executions', {input: input});
+
+async function listActions(z, nextToken) {
+  return makeGetRequest(z, '/actions')
 }
 
-async function getWorkflow(z, workflowId) {
-  return makeGetRequest(z, '/workflows/' + workflowId)
+async function updateAction(z, actionId, body) {
+  return makePatchRequest(z, '/actions/' + actionId, body)
 }
 
-async function listWorkflows(z) {
-  return makeGetRequest(z, '/workflows')
+async function listAgents(z) {
+  return makeGetRequest(z, '/agents')
 }
 
-async function updateWorkflow(z, workflowId, metadata) {
-  return makePatchRequest(z, '/workflows/' + workflowId, {
-    metadata: metadata,
+async function createAgentRun(z, agentId, variables) {
+  return makePostRequest(z, '/agents/' + agentId + '/runs', {
+    variables: variables,
   })
 }
 
-async function getTransition(z, transitionId) {
-  return makeGetRequest(z, '/transitions/' + transitionId)
-}
+async function createDocument(z, inputFileUrl, agentRunId, fileName) {
+  body = {}
+  if (agentRunId !== undefined) {
+    body.agentRunId = agentRunId
+  }
+  if (fileName !== undefined) {
+    body.name = fileName
+  }
 
-async function updateTransition(z, transitionId, parameters) {
-  return makePatchRequest(z, '/transitions/' + transitionId, {
-    parameters: parameters,
-  })
+  const postDocumentsResponse = await makePostRequest(z, '/documents', body)
+  // bundle.inputData.file will be a URL from which we download the file
+  fileResponse = await downloadFile(inputFileUrl, z)
+  const fileServerResponse = await putToFileServer(z, postDocumentsResponse.json.fileUrl, fileResponse.buffer())
```

Comment:
Will errors here be treated in a way that makes the user understand what is wrong? If the document is of a mime type we don't support for instance. And is the file always provided as a URL?